feat: docker build character

thetechnocratic opened the pull request:
# Relates to

easier character builds for docker

# Risks

low

# Background

## What does this PR do?

allows passing a character name in as the second arg to the docker script (also via pnpm scripts)

## What kind of change is this?

QOL improvement for docker

# Documentation changes needed?

updated the docker guide .md file

# Testing

## Detailed testing steps

1. make a character file
2. `pnpm docker:build <character name>
3. `pnpm docker:run <character name>`

## Discord username
thetechnocratic

Saundr21 committed:
Adds initial anyone plugin

HashWarlock commented on Dockerfile:
```diff
@@ -1,5 +1,7 @@
 # Use a specific Node.js version for better reproducibility
 FROM node:23.3.0-slim AS builder
+ARG CHARACTER_PATH=characters/c3po.character.json
```
Is it required to be defined here?

HashWarlock commented:
Should this be able to handle multiple characters to be deployed as well?

thetechnocratic commented:
> Should this be able to handle multiple characters to be deployed as well?

should have no problem building multiple characters, but each `run` command still latches to port 3000, so will need something more intelligent to support multiple deployments at once

mbcse committed:
fix: Added Local Embedding Manager to reuse embed model - Fixes High Ram issue

wtfsayo committed:
Merge branch 'develop' into plugin-arthera-alpha

JoeyKhd committed:
upd: updated validateCharacterConfig to give better feedback

wtfsayo committed:
Merge branch 'develop' into fix/embedding-high-memory-usage

wtfsayo committed:
Merge pull request #1818 from bertux/plugin-arthera-alpha

feat: new plugin Arthera Chain

lalalune committed:
Merge pull request #1951 from JoeyKhd/fix-c3po

fix: fixed error in C3PO and improved error handling feedback

Lukapetro committed:
chore: resolve merge conflicts with develop

wtfsayo committed:
Merge branch 'develop' into plugin-solana-log

wtfsayo committed:
Merge branch 'develop' into pr/1810

conache committed:
Merge remote-tracking branch 'upstream/develop' into plugin-allora

conache committed:
Use topic id as number

kesar committed:
added fraxtal

kesar committed:
added fraxtal

conache committed:
Remove redundant indention

kesar committed:
Merge branch 'develop' into patch-1

wtfsayo committed:
update .env.example and gitignore

wtfsayo committed:
Merge pull request #1810 from KONFeature/1809-add-pglite-adapter

feat: add pglite db adapter

kesar committed:
Merge branch 'develop' into patch-1

wtfsayo committed:
Merge branch 'develop' into pr/1808

ai16z-demirix committed:
Merge branch 'develop' into test/client-twitter

ag-wnl committed:
Merge pull request #1954 from kesar/patch-1

feat: Add fraxtal chain to evm

wtfsayo committed:
Update pnpm-lock.yaml

wtfsayo committed:
Merge branch 'develop' into coinPricePlugin

Lukapetro committed:
Merge branch 'develop' into feature/add-coinmarketcap-plugin

JoeyKhd committed:
upd: send and receiving messages is now correct

wtfsayo committed:
Merge pull request #1808 from proteanx/coinPricePlugin

feat: add coin price plugin (CoinMarketCap, CoinGecko & CoinCap)

JoeyKhd committed:
upd: also display loading chat bubble

Freytes committed:
Merge branch 'develop' into RabbiTrader

JoeyKhd committed:
upd: reworked sidebar and more chat components

shakkernerd committed:
Merge branch 'develop' into test/client-twitter

JoeyKhd committed:
upd: ensure a persistent message cache between navigation

JoeyKhd committed:
upd: improved build config for Vite

azep-ninja committed:
Merge branch 'develop' into feat/seperate-knowledge-multi-agent-rag

wtfsayo committed:
Merge branch 'develop' into pr/1873

wtfsayo committed:
fix invalid response and clean gifs

wtfsayo committed:
Update pnpm-lock.yaml

cygaar committed:
fix: abstract plugin readme

rekpero committed:
fix: tested and fixed deployment and escrow functionality

wtfsayo committed:
Merge pull request #1963 from cygaar/abstract_readme

fix: abstract readme

wtfsayo committed:
Merge branch 'develop' into giphy-plugin

rekpero committed:
fix: update readme and env example for spheron plugin

rekpero committed:
Merge branch 'main' of https://github.com/rekpero/eliza into main

shakkernerd committed:
Merge branch 'develop' into test/client-twitter

Saundr21 committed:
Merge branch 'develop' into feature/anyone-plugin

tcm390 committed:
Merge branch 'develop' into twitter-url

shakkernerd committed:
Merge pull request #1947 from darwintree/twitter-url

feat: improve twitter paragraph splitting if containing url

shakkernerd committed:
Merge branch 'develop' into test/client-twitter

shakkernerd committed:
Merge pull request #1959 from ai16z-demirix/test/client-twitter

test: adding tests for twitter-client

JoeyKhd committed:
upd: work in progress for TTS

JoeyKhd committed:
upd: reset lockfile

JoeyKhd committed:
Merge pull request #1 from JoeyKhd/develop

Develop

0x-ten committed:
Fix db init race condition affecting builds

wtfsayo committed:
Merge branch 'develop' into giphy-plugin

odilitime committed:
Merge branch 'develop' into patch-1

rekpero committed:
fix: fixed the close deployment issue

velvet-shark committed:
Cleaner logs on agent startup

ai16z-demirix committed:
test: changing test structure for core package. Fixing failling outdated tests

ai16z-demirix committed:
test: adding tests for twitter plugin

shakkernerd committed:
Merge branch 'develop' into fix/db-init

guo committed:
Merge branch 'develop' into init-plugin-depin

shakkernerd committed:
Merge branch 'develop' into test/plugin-twitter

JoeyKhd committed:
upd: added proper tooltips

shakkernerd committed:
Merge pull request #1976 from ai16z-demirix/test/plugin-twitter

test: adding tests for twitter plugin

JoeyKhd committed:
upd: added toaster for better error feedbacks

shakkernerd committed:
Merge branch 'develop' into agent-runtime-logging

shakkernerd committed:
Merge pull request #1973 from velvet-shark/agent-runtime-logging

chore: Cleaner logs on agent startup

shakkernerd committed:
Merge branch 'develop' into feature/anyone-plugin

shakkernerd committed:
Merge pull request #1960 from Saundr21/feature/anyone-plugin

feat: anyone plugin to privatise network requests

JoeyKhd committed:
upd: use tooltips

Laurentiu-Andronache committed:
feat: show context when in DEBUG mode

Minco-Yuga committed:
enable all EVM chains by default

Laurentiu-Andronache committed:
fix: many fixes to the Telegram templates

Minco-Yuga committed:
Merge branch 'develop' into develop

JoeyKhd committed:
upd: prevent empty embedding from corrupting the database

JoeyKhd committed:
upd: removed unused import

shakkernerd committed:
Merge pull request #1984 from JoeyKhd/fix-sqlite-error

fix: SQLITE ERROR, zero-lenght vectors not supported

Minco-Yuga committed:
Merge branch 'develop' into develop

JoeyKhd committed:
Merge pull request #2 from JoeyKhd/develop

Develop

ai16z-demirix committed:
test: removing outdated comments

daniel-farina committed:
Merge branch 'develop' into giphy-plugin

shakkernerd committed:
Merge branch 'develop' into fix-telegram

JoeyKhd committed:
upd: chat-tts-button fully operational

JoeyKhd committed:
upd: created API route to allow for proper TTS

JoeyKhd committed:
upd: updated TTS component to use the TTS api route

JoeyKhd committed:
upd: added AI writer effect on incoming messages

tcm390 committed:
Merge pull request #1982 from Laurentiu-Andronache/fix-telegram

fix: many fixes to the Telegram templates

shakkernerd committed:
Merge branch 'develop' into pr-1989

shakkernerd committed:
Merge branch 'develop' into pr-1989

shakkernerd committed:
Merge branch 'develop' into develop

shakkernerd committed:
Merge pull request #1981 from Minco-Yuga/develop

chore: Enable all EVM chains by default

shakkernerd committed:
Merge branch 'develop' into show-context

shakkernerd committed:
Merge pull request #1980 from Laurentiu-Andronache/show-context

chore: add debug logging for context

shakkernerd committed:
Merge branch 'develop' into fix/db-init

shakkernerd committed:
Merge pull request #1968 from 0x-jj/fix/db-init

fix: Fix db init race condition affecting builds

shakkernerd committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge pull request #1966 from rekpero/main

feat: add new plugin - spheron

JoeyKhd committed:
upd: ported full client to vanilla react with swc compiler

shakkernerd committed:
Merge branch 'develop' into pr-1955

JoeyKhd committed:
upd: ensure prettier formatting in entire client folder

shakkernerd committed:
Merge branch 'develop' into fix/embedding-high-memory-usage

shakkernerd committed:
feat: embed types, logs and clean up

shakkernerd committed:
chore: embeddingModelManager -> localEmbeddingModelManager

shakkernerd committed:
Merge branch 'develop' into pr-1943

JoeyKhd committed:
upd: implemented attachments

iankm committed:
Merge branch 'develop' into thirdweb

marcellodesales committed:
:truck: :books: mv README.MD from plugin's src to its root dir

fabianhug committed:
Merge pull request #1418 from iankm/thirdweb

add thirdweb plugin

Ed-Marcavage committed:
Optimizing Technical Docs for LLMs

shakkernerd committed:
Merge branch 'develop' into pr-1934

Ed-Marcavage committed:
Optimizing Technical Docs for LLMs

shakkernerd committed:
fix: ai16z -> elizaos

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
fix: init database

tcm390 committed:
fix dryrun mode in generate tweet and action processing

tcm390 committed:
revert testing code

tcm390 committed:
rename

shakkernerd committed:
chore: build and lint commands

shakkernerd committed:
chore: build and lint config files

shakkernerd committed:
fix: missing init db func call

odilitime committed:
fix type error

odilitime committed:
fix type errors

odilitime committed:
Merge branch 'develop' into hfi

shakkernerd committed:
Merge pull request #1995 from elizaOS/fix/plugin-spheron

fix: build on plugin spheron

shakkernerd committed:
Merge branch 'develop' into hfi

tcm390 committed:
add dry run mode in intercation

tcm390 committed:
Merge branch 'develop' into tcm-fix-dryrun

tomguluson92 committed:
Update lore.md

shakkernerd committed:
Merge pull request #1997 from elizaOS/tcm-fix-dryrun

fix: client twitter dryrun

odilitime committed:
Merge branch 'develop' into hfi

odilitime committed:
Merge pull request #1996 from odilitime/hfi

fix: resolve translation type issue

odilitime committed:
more sane defaults that don't activate things we don't want to activate by default/move format to comment

odilitime committed:
Merge branch 'hfi' of https://github.com/odilitime/eliza into hfi

shakkernerd committed:
Merge branch 'develop' into hfi

shakkernerd committed:
Merge pull request #1999 from odilitime/hfi

fix: fix .env.sample defaults

tcm390 committed:
Merge branch 'develop' into patch-1

Ed-Marcavage committed:
Merge branch 'develop' into fix/jsdoc-automation

shakkernerd committed:
chore: https://ai16z.ai -> https://elizaos.ai

shakkernerd committed:
Merge branch 'develop' into patch-5

shakkernerd committed:
Merge pull request #1998 from tomguluson92/patch-5

chore: update website url in lore.md

shakkernerd committed:
chore: remove conflicting readme files

shakkernerd committed:
Merge branch 'develop' into pr-1621

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge branch 'develop' into pr-1993

shakkernerd committed:
Merge branch 'develop' into pr-1620

shakkernerd committed:
fix: add gaianet embedding check

proteanx committed:
rm legacy vars + update some relevant docs

rms XAI_MODEL & XAI_API_KEY vars

shakkernerd committed:
fix: add memories_768 table

proteanx committed:
expand espanol explanation

shakkernerd committed:
Merge branch 'develop' into patch-1

wtfsayo committed:
Update pnpm-lock.yaml

kush-alloralabs committed:
Update plugins.md -- Allora README Fix

- Fixed a comment-uncomment typo at the end of the Allora Documentation.
- Added a link to more comprehensive documentation + examples at the bottom

shakkernerd committed:
feat: implement getKnowledge, searchKnowledge, createKnowledge, removeKnowledge and clearKnowledge methods

shakkernerd committed:
feat: add KnowledgeSearchRow interface

rferrari committed:
update: Readme and Comments

shakkernerd committed:
Merge pull request #2005 from elizaOS/feat/adapter-pglite

feat: implement getKnowledge, searchKnowledge, createKnowledge, removeKnowledge and clearKnowledge methods

shakkernerd committed:
Merge branch 'develop' into patch-1

shakkernerd committed:
Merge pull request #1902 from hazelnutcloud/patch-1

chore: Make `generateNewTweet` public

shakkernerd committed:
Merge branch 'develop' into rmLegacyVars

shakkernerd committed:
Merge pull request #2001 from proteanx/rmLegacyVars

fix: remove legacy variables (XAI_MODEL,  XAI_API_KEY & IMAGE_GEN)

shakkernerd committed:
Merge branch 'develop' into patch-1

HashWarlock committed:
Fix Auto Client bug

shakkernerd committed:
Merge pull request #2003 from kush-alloralabs/patch-1

Update plugins.md -- Allora README Fix

wtfsayo committed:
Merge branch 'develop' into plugin-solana-log

shakkernerd committed:
Merge branch 'develop' into pr-2006

wtfsayo committed:
Merge branch 'develop' into plugin-solana-log

shakkernerd committed:
Merge branch 'develop' into bug/fix-auto-client

shakkernerd committed:
Merge pull request #2007 from elizaOS/bug/fix-auto-client

fix: Auto Client bug

wtfsayo committed:
Merge branch 'develop' into plugin-solana-log

wtfsayo committed:
Merge pull request #1888 from sinecose/plugin-solana-log

chore(plugin-solana): replace console with elizaLogger

shakkernerd committed:
Merge branch 'develop' into pr-1895

shakkernerd committed:
Merge branch 'develop' into pr-1895

shakkernerd committed:
Merge branch 'develop' into pr-1874

koofree committed:
fix: packages/adapter-postgres/schema.sql isShared needs double quote

shakkernerd committed:
Merge branch 'develop' into develop

shakkernerd committed:
Merge pull request #2008 from koofree/develop

fix: packages/adapter-postgres/schema.sql isShared needs double quote

kristjanpeterson1 committed:
Update the Galadriel LLM integration to reflect their product updates

Lukapetro committed:
chore: resolve merge conflicts with develop

shakkernerd committed:
Merge branch 'develop' into pr-1876

shakkernerd committed:
chore: extra instructions for twitter approval configs

shakkernerd committed:
chore: pnpm lock file

Lukapetro committed:
Merge branch 'develop' into feature/add-coinmarketcap-plugin

shakkernerd committed:
Merge branch 'develop' into pr-1873

fabianhug committed:
fix: updated request body structure, error handling, JSON validation and Cloudflare response checks in `OpacityAdapter`

fabianhug committed:
Merge branch 'develop' into plugin-opacity

jonathanykh committed:
chore(ci): add pnpm lockfile consistency check workflow

fabianhug committed:
Merge pull request #1673 from RonTuretzky/plugin-opacity

Opacity Verifiable Interference zkTLS Plugin

wtfsayo committed:
remove-near-solana-slippage variable conflict

dto-simba committed:
add tg api url config in env

wtfsayo committed:
Update plugin readme

wtfsayo committed:
Merge branch 'develop' into feature/add-coinmarketcap-plugin

shakkernerd committed:
Merge pull request #2016 from elizaOS/fix-near-slippage-env-conflict

fix: remove near-solana slippage env conflict

shakkernerd committed:
Merge branch 'develop' into develop

wtfsayo committed:
Merge branch 'develop' into feature/add-coinmarketcap-plugin

wtfsayo committed:
Merge branch 'develop' into pr/2014

shakkernerd committed:
Merge pull request #2011 from kristjanpeterson1/develop

feat: Update the Galadriel LLM integration to reflect their product updates

wtfsayo committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge branch 'develop' into develop

wtfsayo committed:
remove character has `.exe` in name lol

shakkernerd committed:
Merge pull request #2015 from jonathanykh/develop

chore(ci): add pnpm lockfile consistency check workflow

wtfsayo committed:
Merge branch 'develop' into feature/add-coinmarketcap-plugin

shakkernerd committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge pull request #2014 from dto-simba/main

feat: add telegram api root config in env

shakkernerd committed:
Merge branch 'develop' into pr-1868

wtfsayo committed:
Merge branch 'develop' into RabbiTrader

wtfsayo committed:
upgrade package names

wtfsayo committed:
Merge branch 'develop' into feature/add-coinmarketcap-plugin

wtfsayo committed:
remove prettier config

wtfsayo committed:
fix letzAI issues

wtfsayo committed:
fix lint issues

JoeyKhd committed:
upd: audio-recorder operational

shakkernerd committed:
Merge pull request #2021 from elizaOS/fix-letzAI-build-issues

fix: letzAi build issues

wtfsayo committed:
Merge branch 'develop' into pr/1785

wtfsayo committed:
Merge pull request #1785 from Freytes/RabbiTrader

feat: plugin new RabbiTrader

wtfsayo committed:
Merge branch 'develop' into feature/add-coinmarketcap-plugin

wtfsayo committed:
Merge pull request #1773 from Lukapetro/feature/add-coinmarketcap-plugin

Feature/add coinmarketcap plugin

WAGMIBRO committed:
Update README_RU.md

typo fix

wtfsayo committed:
remove defillama plugin

JoeyKhd committed:
upd: added connection status

wtfsayo committed:
final: fix rabbi trader

shakkernerd committed:
chore: fix build on plugin rabbi trader

shakkernerd committed:
Merge pull request #2027 from elizaOS/remove-defillama-plugin

fix: remove defillama plugin & improve rabbi trader

JoeyKhd committed:
upd: bump client to react 19

JoeyKhd committed:
upd: improved connection status component

JoeyKhd committed:
upd: resolved hydration errors

ag-wnl committed:
Merge branch 'develop' into patch-1

ag-wnl committed:
Merge pull request #2026 from WAGMIBRO/patch-1

Update README_RU.md

JoeyKhd committed:
upd: added more precise refetching and resolved an issue with the TTS player not displaying stop properly

brauliolomeli committed:
fix: include schema.sql and seed.sql in package files

antman1p committed:
Update .env.example

Added Supabase variables.

tcm390 committed:
Merge branch 'develop' into patch-2

tcm390 committed:
Merge pull request #2031 from antman1p/patch-2

Update .env.example

JoeyKhd committed:
upd: ensure file gets uploaded properly

JoeyKhd committed:
fix: resolved build errors

JoeyKhd committed:
upd: added update component to frontend displaying that user should update his app

JoeyKhd committed:
upd: sidebar now has active state

odilitime committed:
Merge pull request #2030 from GravitonINC/main

fix: include schema.sql and seed.sql in package files

odilitime committed:
Merge branch 'develop' into bugfix/move-web-search-plugin-README-to-root-dir

HashWarlock committed:
fix: use the salt with path for reliable key derivation. clarify subject is for certificate chain

odilitime committed:
Merge pull request #1992 from marcellodesales/bugfix/move-web-search-plugin-README-to-root-dir

fix: 🚚 📚 move plugin-web-search readme to root dir

JoeyKhd committed:
upd: added gaps

HashWarlock committed:
feat: add unit tests for TEE plugin

JoeyKhd committed:
upd: updated version.sh to have 0 dependencies

shakkernerd committed:
Merge branch 'develop' into pr-2038

AnonJon committed:
fix: update slack client readme manifest (#2044)

HashWarlock committed:
Merge branch 'develop' into fix/tee-derive-key

shakkernerd committed:
Merge pull request #2039 from Phala-Network/fix/tee-derive-key

fix: Update Key Derive in TEE

shakkernerd committed:
Revert "fix: Update Key Derive in TEE"

shakkernerd committed:
Merge pull request #2049 from elizaOS/revert-2039-fix/tee-derive-key

Revert "fix: Update Key Derive in TEE"

wtfsayo committed:
Merge branch 'develop' into feature/docker-build-character
